src/mds/Server.cc

```diff
@@ -5534,7 +5534,14 @@ void Server::handle_client_unlink(MDRequestRef& mdr)
   if (in->is_dir() && in->has_subtree_root_dirfrag()) {
     // subtree root auths need to be witnesses
     set<mds_rank_t> witnesses;
-    in->list_replicas(witnesses);
+
+    list<CDir*> dfls;
+    diri->get_subtree_dirfrags(dfls);
+    for (auto p : dfls) {
+      if (mds->get_nodeid() != p->get_dir_auth().first)
+	witnesses.insert(p->get_dir_auth().first);
+    }
```

Comment:
If we don't make other replicas a witness here, at which point do they find out the inode has been unlinked?